patch for block-wise quantization + pt2e

cccclai opened the pull request:
Summary: As title, needed for enable qcom block-wise quantization kernel

Test Plan: local test

Differential Revision: D67985303

cccclai commented:
@pytorchbot merge

cccclai commented:
@pytorchbot rebase

jerryzh168 commented on torch/ao/quantization/quantize_pt2e.py:
this file already import from constant_folding which will register the op, so the import in this file should be removed I think

cccclai committed:
patch for block-wise quantization + pt2e (#144492)

Summary:

As title, needed for enable qcom block-wise quantization kernel

Test Plan: local test

Reviewed By: angelayi, billmguo

Differential Revision: D67985303

huydhn commented:
@cccclai I think this change breaks ExecuTorch test 

[GH job link](https://github.com/pytorch/pytorch/actions/runs/12822518950/job/35755934577) [HUD commit link](https://hud.pytorch.org/pytorch/pytorch/commit/1d43b8150852cdfcbe754edcf027d6e25f33ac63)

[GH job link](https://github.com/pytorch/pytorch/actions/runs/12826561430/job/35767397037) [HUD commit link](https://hud.pytorch.org/pytorch/pytorch/commit/dbed747aae223d53ca4e22fe45c24d1d9a8b4432)

As this has been reverted and the revert commit is now part of diff train, I think it's best to close this and create a new PR to fix the issue and reland the change

cc @kit1980

andrewor14 commented:
Closing this in favor of #146946
